ui/app/components/auth/form/oidc-jwt.ts

```diff
@@ -62,188 +58,193 @@ export default class AuthFormOidcJwt extends AuthBase {
   _formData: FormData = new FormData();
 
   // set during auth prep and login workflow
-  @tracked fetchedRole: RoleJwtModel | null = null;
+  @tracked authUrl: string | null = null;
   @tracked errorMessage = '';
   @tracked isOIDC = true;
 
-  get tasksAreRunning() {
-    return this.prepareForOIDC.isRunning || this.exchangeOIDC.isRunning;
-  }
-
   get icon() {
-    return this?.fetchedRole?.providerIcon || '';
+    // Right now there is a bug in HDS where the name includes a space, this line can be removed when we
+    // upgrade to an HDS version with the corrected icon name
+    if (this.provider === 'Ping Identity') return 'ping-identity ';
+    return this.provider ? dasherize(this.provider.toLowerCase()) : '';
   }
 
   get providerName() {
-    return `with ${this?.fetchedRole?.providerName || 'OIDC Provider'}`;
+    return `with ${this.provider || 'OIDC Provider'}`;
+  }
+
+  get provider() {
+    const { hostname } = parseURL(this.authUrl) as UrlParseData;
```

Comment:
yes..typecasting is lazy here but updating parseURL to typescript felt out of scope since this PR is already massive..